primitives/src/smart_extension.rs

```diff
@@ -0,0 +1,28 @@
+use codec::{Decode, Encode};
+use sp_std::prelude::Vec;
+
+/// Smart Extension types
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, PartialOrd, Ord)]
+pub enum SmartExtensionTypes {
+    TransferManager,
+    Offerings,
+    Custom(Vec<u8>),
+}
+
+impl Default for SmartExtensionTypes {
+    fn default() -> Self {
+        SmartExtensionTypes::Custom(b"undefined".to_vec())
+    }
+}
+
+#[derive(Encode, Decode, Default, Clone, PartialEq, Debug)]
+pub struct SmartExtension<U> {
```

Comment:
If `U` is an account ID type, I think it would be better to call it that way, `AccountId`. In general Substrate follows best practices of having descriptive type parameter names.

Reply:
The main reason behind following that approach is because I am not able to access the `frame_system` trait AccountId here. Which has different types of what we are declaring in `lib.rs` in primitives that cause type mismatch. that's why I settle with an associated type.

Reply:
Using the type parameter makes sense. I'm only commenting about the possibility to make code self-documenting by using a descriptive name for this type parameter.